### DISCUSSION TITLE
Not getting any data in pair array

### DISCUSSION THREAD

**Discussion by VedikaGujar-GSLab** [dodona-edu/dolos#1063] [Q&A] [answered]
Hi All,

I am using below code
```
const dolos = require("@dodona/dolos-lib");
const instance = new dolos.Dolos();
const report = instance.analyzePaths(files);

report.then(
        function(value) {             
           console.log( JSON.stringify(value));
         },
        function(err) {   
            console.log(err);
        }
      );
```

I am getting response.

<img width="222" alt="image" src="https://user-images.githubusercontent.com/61533672/220312428-f0c56db7-e619-4308-b937-b275cd9d30d7.png">

where pairs is null.

Before when i was using Dolos, i used to get below response

<img width="404" alt="image" src="https://user-images.githubusercontent.com/61533672/220312724-e2b65a0f-8194-4d0a-8a67-65c4ba6b9940.png">

**Comment by rien** [comment, marked as answer]
Yes, in version 2.1.0 we've changed how we calculate pairs in order to reduce memory usage. We no longer have a list of `scored` pairs in the `Report` object that is the result of `analyzePaths`. Instead, you can call the `allPairs()` function. You can change your code to this:

```javascript
const dolos = require("@dodona/dolos-lib");
const instance = new dolos.Dolos();
const reportPromise = instance.analyzePaths(files);

reportPromise.then(
  function(report) {
    const pairs = report.allPairs();
    console.log( JSON.stringify(pairs));
  },
  function(err) {
    console.log(err);
  }
);
```
You could also set `report.scored = report.allPairs();` if you want to do minimal changes to your code.

There are other small breaking changes like this because of the update, you can check them out here: https://github.com/dodona-edu/dolos/pull/1011
